Opentrons/opentrons
# feat(api): Explicit cmdline and jupyter entrypoints

Create explicit entrypoints for commandline and jupyter execution of protocols, and refactor protocol parse and execution to pull more logic out of the rpc layer.

There's a new `opentrons_execute` entrypoint that would only be valid on a robot, and a `get_protocol_api` method that can be used from jupyter. `opentrons_execute` works regardless of protocol api version.

## Testing
- Install and make sure the entrypoints work (unless you install the buildroot system build, you won't have the `opentrons_execute` binding but you can still run it with `python -m opentrons.execute)`
- Run python and json protocols to check that the refactors didn't break anything

Closes #3755, closes #3546
